keycloak/keycloak-admin-ui
# User fed cypress tests should not fail if there are existing providers on the system

## Motivation
Fixes https://github.com/keycloak/keycloak-admin-ui/issues/376

## Brief Description
The tests for kerberos and ldap providers both assumed that the system would be clean and no previously configured providers would be on the system. This resulted in the tests failing if someone were to run the tests locally and have providers already existing on their system. Now the tests will create the initial provider card from the card view if providers exist; otherwise, it will create it from the empty state card.

This fix also includes some major refactoring of the user fed provider tests, most notably, reducing code redundancy and consolidating the kerberos and ldap page objects into one single page object to be used by both kerberos and ldap tests.

## Verification Steps
1. Go to User Federation and create one or more kerberos providers, if at least one does not already exist.
2. Start cypress and run the user_fed_kerberos_test.
3. Verify that the tests pass and the existing provider(s) from step 1 still exist on the system.
4. Delete all existing providers and re-run the test.
5. Verify that the tests pass, and there are no providers configured on the system.
6. Repeat steps 1-5 for LDAP.

## Checklist:

## Additional Notes
Note that the original issue was filed agains kerberos test but this fixes the issue for both kerberos and ldap tests.
